kagen/generators/brain/brain.cpp

```diff
@@ -0,0 +1,249 @@
+#include "kagen/generators/brain/brain.h"
+
+#include "kagen/context.h"
+#include "kagen/kagen.h"
+
+#include <algorithm/AlgorithmEnum.h>
+#include <mpi.h>
+
+#include "algorithm/BarnesHutInternal/BarnesHutCell.h"
+#include "algorithm/BarnesHutInternal/BarnesHutInvertedCell.h"
+#include "algorithm/Kernel/Kernel.h"
+#include "algorithm/NaiveInternal/NaiveCell.h"
+#include "neurons/CalciumCalculator.h"
+#include "neurons/helper/SynapseDeletionFinder.h"
+#include "neurons/input/BackgroundActivityCalculators.h"
+#include "neurons/input/FiredStatusCommunicationMap.h"
+#include "neurons/input/SynapticInputCalculator.h"
+#include "neurons/input/SynapticInputCalculators.h"
+#include "neurons/models/NeuronModels.h"
+#include "neurons/models/SynapticElements.h"
+#include "sim/Essentials.h"
+#include "sim/Simulation.h"
+#include "sim/random/SubdomainFromNeuronPerRank.h"
+#include "structure/Partition.h"
+#include <memory>
+namespace kagen {
+PGeneratorConfig
+BrainFactory::NormalizeParameters(PGeneratorConfig config, PEID, const PEID size, const bool /*output*/) const {
+    config.brain.num_neurons_per_rank = (config.n + size - 1) / size;
+    return config;
+}
+std::unique_ptr<Generator>
+BrainFactory::Create(const PGeneratorConfig& config, const PEID rank, const PEID size) const {
+    return std::make_unique<BrainGenerator>(config, rank, size);
+}
+
+BrainGenerator::BrainGenerator(const PGeneratorConfig& config, const PEID rank, const PEID size)
+    : _sim(std::make_unique<Essentials>(), std::shared_ptr<Partition>(nullptr)),
+      _config(config.brain),
+      _rank(rank) {
+    MPIWrapper::init(1, nullptr);
```

Comment:
does this  call MPI_Init? wouldn't that crash? 

the "KaGen" way would be to use Finalize() for everything that requires communication (the communicator is only passed to Finalize()), but obviously, this is stroingly biased by the communication free nature of KaGen. 

Should we generate the entire graph there or keep it as it? Personally I don't have a strong opinion on whether we should stick to this model or not